FSD-3  Algorithm Concepts

AMubinshaikh opened the pull request:
FSD-3 is based on Algorithm concepts like
Sorting
Searching & recursive
Bubble sort with example of ascending order and descending

QuickSort
Binary Search with and without recursive function

AMubinshaikh committed:
FSD-3 is based on ALgorithm Concepts

FSD-3 is based on Algorithm concepts like
Sorting
Searching & recursive
Bubble sort with example of ascending order and descending

QuickSort
Binary Search with and without recursive function

abdulmuheet commented on Algorithms_Concepts/Factorial.cs:
```diff
@@ -0,0 +1,33 @@
+        public int CalculateFactorial( int number)
+        {
+           if(number==0 || number==1 )
+           {
```
curly braces are nor required for a single line of code in it.

AMubinshaikh replied:
done

abdulmuheet commented on Algorithms_Concepts/Factorial.cs:
```diff
@@ -0,0 +1,33 @@
+            result = CalculateFactorial(number);
+            Console.WriteLine("Factorial = "+ result);
+        }
+        public int CalculateFactorial( int number)
```
remove space before int

abdulmuheet commented on Algorithms_Concepts/BinarySearch.cs:
```diff
@@ -0,0 +1,43 @@
+        int[] array = new int[100];
+        public void AcceptArray()
+        {
+               Console.WriteLine ("Define Array size");
```
Instead of getting input from the console, get it in method input parameters.

abdulmuheet replied:
This is applicable to all the places.

abdulmuheet commented on Algorithms_Concepts/BinarySearch.cs:
```diff
@@ -0,0 +1,43 @@
+                   array[arrayLoop] = Convert.ToInt32 (Console.ReadLine ());
+               }
+        }
+          public int BinarySearchMethod (int[] arr, int min, int max, int search) {
```
'Method' word is not required.

abdulmuheet replied:
give better variable name.

abdulmuheet commented on Algorithms_Concepts/BinarySearchRecursive.cs:
```diff
@@ -0,0 +1,52 @@
+/*
```
What is the use of this class? all code is commented.

abdulmuheet commented on Algorithms_Concepts/BubbleSort.cs:
```diff
@@ -0,0 +1,49 @@
+        public void SortArray()
+        {
+            
+
```
extra line break not required.

abdulmuheet commented on Algorithms_Concepts/BubbleSort.cs:
```diff
@@ -0,0 +1,49 @@
+
+           for (arrayLoop = 0; arrayLoop < arraySize; arrayLoop++) 
+           {
+              for (int inetrnalLoop = 0; inetrnalLoop < arraySize - 1; inetrnalLoop++) 
```
typo, also rename inetrnalLoop -> 'inner' and the above loop variable to 'outer'

abdulmuheet commented on Algorithms_Concepts/Factorial.cs:
```diff
@@ -0,0 +1,33 @@
+       public int number;
+        public void AcceptNumber()
+        {
+            
```
extra line space should be removed from all the places.

abdulmuheet commented on Algorithms_Concepts/Factorial.cs:
```diff
@@ -0,0 +1,33 @@
+        public void DisplayFactorial()
+        {
+            int result;
+            result = CalculateFactorial(number);
```
directly declare the variable here.

abdulmuheet commented on Algorithms_Concepts/Program.cs:
```diff
@@ -0,0 +1,70 @@
+            
+            //---------------------------
+
+
```
remove extra line spaces.

abdulmuheet commented on Algorithms_Concepts/QuickSort.cs:
```diff
@@ -0,0 +1,72 @@
+{
+    public class QuickSort
+    {
+         int arrayLoop;
```
The indentation is not correct.

abdulmuheet commented on Algorithms_Concepts/QuickSort.cs:
```diff
@@ -0,0 +1,72 @@
+                    right--;
+                }
+
+                if (left != right) {
```
add a comment to explain the code wherever some logic is written.

abdulmuheet commented on Algorithms_Concepts/bin/Debug/netcoreapp3.1/Algorithms_Concepts.deps.json:
```diff
@@ -0,0 +1,23 @@
+{
```
I don't think this file is required to be in source control.

abdulmuheet commented on Algorithms_Concepts/QuickSort.cs:
```diff
@@ -0,0 +1,72 @@
+            sort (0, len - 1);
+        }
+        public void sort (int left, int right) {
+            int pivot, leftend, rightend;
```
use better variable names

abdulmuheet commented on Algorithms_Concepts/QuickSort.cs:
```diff
@@ -0,0 +1,72 @@
+       
+        public void DisplaySortedArray()
+        {
+            for (int j = 0; j < arraySize; j++) 
```
j is not a good name ;)

abdulmuheet commented on Algorithms_Concepts/Program.cs:
```diff
@@ -0,0 +1,70 @@
+    {
+        static void Main(string[] args)
+        {
+            //Bubble sot Ascending Order
```
you can uncomment all these codes and pass input array/values in the function.

abdulmuheet commented on Algorithms_Concepts/BubleSortDescending.cs:
```diff
@@ -0,0 +1,49 @@
+            }
+        }
+
+        public void PrintSortArray()
```
Printing should be done on the client side, in program.cs

abdulmuheet commented on Algorithms_Concepts/BinarySearch.cs:
```diff
@@ -0,0 +1,43 @@
+     int arrayLoop;
+        int arraySize;
+        int search;
+        int[] array = new int[100];
```
rename array -> numbers

abdulmuheet reviewed (changes requested):
@AMubinshaikh Please refer to the comment mentioned inline.

AMubinshaikh committed:
FSD 3 changes are done

All changes are done provided in PR's comment. please re-check.

AMubinshaikh committed:
Delete Algorithms_Concepts.runtimeconfig.json

AMubinshaikh committed:
Delete project.nuget.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.csproj.FileListAbsolute.txt

AMubinshaikh committed:
Delete Algorithms_Concepts.runtimeconfig.dev.json

AMubinshaikh committed:
Delete Algorithms_Concepts.csproj.CoreCompileInputs.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.csproj.nuget.g.targets

AMubinshaikh committed:
Delete project.assets.json

AMubinshaikh committed:
Delete Algorithms_Concepts.genruntimeconfig.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.AssemblyInfoInputs.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.csproj.nuget.dgspec.json

AMubinshaikh committed:
FSD-3 all changes done

Changing Function name.

AMubinshaikh committed:
Changes are done

According to provided comments all changes are done
Change variables name
added comments
indentation

AMubinshaikh committed:
Update Program.cs

AMubinshaikh committed:
Update BinarySearch.cs

AMubinshaikh committed:
Delete Algorithms_Concepts.deps.json

AMubinshaikh committed:
Delete Algorithms_Concepts.AssemblyInfo.cs

AMubinshaikh committed:
Delete Algorithms_Concepts.csproj.nuget.g.props

AMubinshaikh committed:
Delete .NETCoreApp,Version=v3.1.AssemblyAttributes.cs

AMubinshaikh committed:
Delete Algorithms_Concepts.pdb

AMubinshaikh committed:
Delete Algorithms_Concepts.exe

AMubinshaikh committed:
Delete Algorithms_Concepts.dll

AMubinshaikh committed:
Delete Algorithms_Concepts.csprojAssemblyReference.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.assets.cache

AMubinshaikh committed:
Delete Algorithms_Concepts.pdb

AMubinshaikh committed:
Delete Algorithms_Concepts.exe

AMubinshaikh committed:
Delete Algorithms_Concepts.dll

abdulmuheet commented on Algorithms_Concepts/BubleSortDescending.cs:
```diff
@@ -0,0 +1,19 @@
+   public class BubbleSortDescending {
+      public void SortDescending (int[] numbers) {
+         for (int arrayLoop = 0; arrayLoop < numbers.Length; arrayLoop++) {
+            for (int inetrnalLoop = 0; inetrnalLoop < numbers.Length - 1; inetrnalLoop++) {
```
typo, please correct it.

abdulmuheet replied:
inetrnalLoop it should be internalLoop

abdulmuheet reviewed (approved):
approved
